lms/resources/_js_config.py

```diff
@@ -64,19 +87,34 @@ def _debug(self):
         return debug_info
 
     @property
-    def _hypothesis_config(self):
+    @functools.lru_cache()
+    def _hypothesis_client(self):
         """
-        Return the Hypothesis client config object for the current request.
-
-        See: https://h.readthedocs.io/projects/client/en/latest/publishers/config/#configuring-the-client-using-json
-
+        Return the config object for the Hypothesis client.
+
+        This is a cached property (with @functools.lru_cache()) so that it's
+        mutable. Code outside of this method can do
+        self._hypothesis_client["foo"] = "bar" and, because
+        self._hypothesis_client always returns the same cached dict and not a
+        new dict every time, the mutation will be preserved.
+
+        This is implemented as a property rather than having the __init__()
+        method initialize the self._hypothesis_client dict as a simple
+        attribute, because generating the dict can raise. If it's going to
+        raise we don't want that to happen until as late as possible (not until
+        someone actually tries to access the dict). If JSConfig.__init__() were
+        to raise that would change the location of the crash in Pyramid's
+        request processing pipeline and could change the type of error response
+        as a result.
```

Comment:
These docstrings are a bit verbose but the reason for why this needs to be a cached property and not just either a method or `self` attr initialized in `__init__()` is quite obscure so I thought it worth spelling out

Reply:
Yeah, it's kind of a copy of the comment above. Similar thoughts on this one. I think a little more thought could slim this down. Or maybe a note at the top about the cached properties, or a decorator to house the string. 

Doesn't really matter but I always groan when I see a wall of text like, because it's _probably_ not relevant to what I'm up to right now, but I'd have to read the thing every time to be sure.